### PR TITLE
[node-cron] Schedule supports promises

### PR DESCRIPTION
**Pull request by jacob-hd**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: **See justification below**
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

Changes
------------
It is clear from the `node-cron` code that the `func` parameter of `schedule` **already** supports both `() => void` and `() => Promise<void>` (`async` function and/or function that returns promise).

For proof, please see:
https://github.com/node-cron/node-cron/blob/master/src/task.js#L52-L54

Here, `execution` comes from `func`:
https://github.com/node-cron/node-cron/blob/master/src/node-cron.js#L43

### CODE REVIEW

**Review by riki-orange — APPROVED** [review]

File: types/node-cron/index.d.ts [hunk 1]
```diff
@@ -4,7 +4,11 @@
 //                 burtek <https://github.com/burtek>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-export function schedule(cronExpression: string, func: () => void, options?: ScheduleOptions): ScheduledTask;
+export function schedule(
+    cronExpression: string,
+    func: () => void | Promise<void>,
```
Comment (SimonSchick) [inline types/node-cron/index.d.ts]: Technically `any` return value is valid here, I agree with keeping it `void` tho as this enforced clean code :)

**Review by SimonSchick — APPROVED** [review]

File: types/node-cron/index.d.ts (hunk 1 above)
Comment (burtek) [reply types/node-cron/index.d.ts]: Doesn't `() => void` cover the `async` case already?

Comment (burtek) [inline types/node-cron/index.d.ts]: Doesn't `() => void` cover the `async` case already? According to [Typescript's Do's and Don'ts](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#callback-types) `void` return value should be used for callbacks whose value will be ignored. I don't really see any advantage of adding `Promise` to the signature here...

Comment (burtek) [reply types/node-cron/index.d.ts]: According to [Typescript's Do's and Don'ts](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#callback-types) `void` return value should be used for callbacks whose value will be ignored.

Comment (SimonSchick) [reply types/node-cron/index.d.ts]: That's what I meant :)

Comment (jacob-hd) [reply types/node-cron/index.d.ts]: You know what, you're totally right! lol

Closing this PR. Thanks

Comment (burtek) [reply types/node-cron/index.d.ts]: In fact, even if Promises weren't supported explicitly, they would still workas usual ;)

**Pull request closed without merging**